src/threads_arg/threads_to_vcf.py

```diff
@@ -1,10 +1,34 @@
 from .serialization import load_instructions, load_metadata, load_sample_names
+from .utils import read_variant_metadata
 from threads_arg import VCFWriter
 
-def threads_to_vcf(threads):
+def threads_to_vcf(threads, samples=None, variants=None):
+    if samples is None:
+        try:
+            sample_names = load_sample_names(threads)
+        except KeyError:
+            raise RuntimeError("Unable to load sample information from threading instructions. This may because the input was inferred using an older version of Threads or without the --save_metadata flag. Please provide files with variant information (in .bim/.pvar format) and sample IDs in a file with one sample per line.")
+    else:
+        with open(samples, "r") as samplefile:
+            sample_names = [l.strip() for l in samplefile]
+
+    if variants is None:
+        try:
+            variant_metadata = load_metadata(threads)
+        except KeyError:
+            raise RuntimeError("Unable to load sample information from threading instructions. This may because the input was inferred using an older version of Threads or without the --save_metadata flag. Please provide files with variant information (in .bim/.pvar format) and sample IDs in a file with one sample per line.")
```

Comment:
Should this be "Unable to load variant information" rather than "...sample information"?